src/daft-distributed/src/pipeline_node/translate.rs

```diff
@@ -224,8 +224,12 @@ impl TreeNodeVisitor for LogicalPlanToPipelineNodeTranslator {
             LogicalPlan::SubqueryAlias(_)
             | LogicalPlan::Union(_)
             | LogicalPlan::Intersect(_)
-            | LogicalPlan::Shard(_) => {
-                panic!("Logical plan operators SubqueryAlias, Union, Intersect, and Shard should be handled by the optimizer")
+            | LogicalPlan::Shard(_)
+            | LogicalPlan::Offset(_) => {
```

Comment:
Instead of creating an `Offset` operator, I'd prefer to instead have a `Slice` operator that takes an offset and an optional length: `Slice(offset, Some(length))`

This would allow us to eventually remove the `Limit` operator and replace it with the `Slice` operator. This would make it easier to push down both offset and limit to sources instead of needing to pushdown both separately.